### PR TITLE
Add a WebView method to clear cookies on Mac and Windows

### PR DESCRIPTION
**Pull request by awood314**
TSIA

### CODE REVIEW

File: gui/choc_WebView.h [hunk 1]
```diff
@@ -1097,7 +1117,21 @@ MIDL_INTERFACE("15e1c6a3-c72a-4df3-91d7-d097fbec6bfd")
 ICoreWebView2PermissionRequestedEventHandler : public IUnknown
 {
 public:
-     virtual HRESULT STDMETHODCALLTYPE Invoke(ICoreWebView2*, ICoreWebView2PermissionRequestedEventArgs*) = 0;
+    virtual HRESULT STDMETHODCALLTYPE Invoke(ICoreWebView2*, ICoreWebView2PermissionRequestedEventArgs*) = 0;
```
Comment (steve-mackinnon) [inline gui/choc_WebView.h]: non-blocking: this deleted space can be added back

**Review by steve-mackinnon — APPROVED** [review]
LGTM!

File: gui/choc_WebView.h [hunk 2]
```diff
@@ -520,6 +523,17 @@ struct choc::ui::WebView::Pimpl
     bool loadedOK() const           { return getViewHandle() != nullptr; }
     void* getViewHandle() const     { return (CHOC_OBJC_CAST_BRIDGED void*) webview; }
 
+    bool clearCookies()
+    {
+        auto store = objc::call<id> (objc::getClass ("WKWebsiteDataStore"), "defaultDataStore");
+        auto allTypes = objc::call<id> (objc::getClass ("WKWebsiteDataStore"), "allWebsiteDataTypes");
+        objc::call<void>( store, "fetchDataRecordsOfTypes:completionHandler:", allTypes, ^(id records)
+            {
+                objc::call<void> (store, "removeDataOfTypes:forDataRecords:completionHandler:", allTypes, records, ^(void){});
```
Comment (nick-thompson) [inline gui/choc_WebView.h:532]: Is there any reason we might need to wait for that completion handler callback?

**Review by nick-thompson — APPROVED** [review]
Nice one!

File: gui/choc_WebView.h (hunk 2 above)
Comment (awood314) [reply gui/choc_WebView.h:532]: I guess technically the cookies won't be cleared until that callback is called, though we don't get any confirmation from it that anything was successful. do you think it's worth adding a callback to `clearCookies`?

File: gui/choc_WebView.h (hunk 1 above)
Comment (awood314) [reply gui/choc_WebView.h]: ah, weird. this was a tab and a space, but I guess that's upstream 🤷

File: gui/choc_WebView.h (hunk 2 above)
Comment (nick-thompson) [reply gui/choc_WebView.h:532]: Hmm I suppose the only real issue would be a race condition if the caller thinks cookies are cleared but they haven't actually been cleared yet. It seems like super low likelihood... I think it's probably fine

**Pull request merged**